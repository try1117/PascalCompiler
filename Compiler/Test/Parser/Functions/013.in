program test;

function a(i: integer): integer;

  procedure aa;
  begin
    exit;
  end;

  procedure ab;
  begin
    exit;
  end;

begin
  aa;
  exit(5);
  ab;
end;

begin
end.
