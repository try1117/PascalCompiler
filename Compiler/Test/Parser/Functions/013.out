test : function()
   resultType : Nil

test declarations:
   a : function(
      i : Integer
   ) resultType : Integer

   a declarations:
      aa : function()
         resultType : Nil

      aa declarations:
      |-- Statements
      |            |-- exit

      ab : function()
         resultType : Nil

      ab declarations:
      |-- Statements
      |            |-- exit

   |-- Statements
   |            |-- Call aa
   |            |-- exit
   |            |      |-- 5
   |            --- Call ab

